modules/dhcp/providers/server/isc.rb

```diff
@@ -78,13 +78,13 @@ def parse_config_and_leases_for_records
             opts.merge!(parse_record_options(data))
           end
         end
-        begin
-          subnet = @service.find_subnet(opts[:ip])
-          next unless subnet
-          ret_val << Proxy::DHCP::Reservation.new(opts.merge(:subnet => subnet))
-        rescue Exception => e
-          logger.warn "skipped #{hostname} - #{e}"
+        if opts[:deleted]
+          ret_val.delete_if { |rec| rec.options[:hostname] == hostname }
+          next
         end
```

Comment:
The leases file has to be parsed in order; a deleted statement will apply to earlier entries, and then subsequent entries should be allowed through. so...yeah, actually a hash by hostname should be fine. I actually can't believe I never noticed that deleted entries weren't being parsed. This has been driving my team crazy for... two years?